src/txrebroadcast.cpp

```diff
@@ -96,3 +104,24 @@ void TxRebroadcastHandler::RecordAttempt(indexed_rebroadcast_set::index<index_by
 
     m_attempt_tracker.modify(entry_it, UpdateRebroadcastEntry);
 };
+
+void TxRebroadcastHandler::TrimMaxRebroadcast()
```

Comment:
Entries from `m_attempt_tracker` seem to be only cleared after 3 months or when it's full. Shouldn't we also remove them if the tx is removed from the mempool because it has made it into a block (which would seem like the most common cause for removing)?

Reply:
Removing entries from `m_attempt_tracker` when tx is removed from mempool doesn't seem necessary given `m_attempt_tracker` isn't used as a source for txs that need to be rebroadcasted.

I also don't see an issue with keeping txs that have already been included in a block or RBFed out of mempool in the `m_attempt_tracker` for up to 90 days (as per current setting) or till it gets removed after 500 limit is reached. Removing these entries from `m_attempt_tracker` would  add unnecessary complexity for a small benefit of a little saved memory as far as I understand. On that note, would perhaps removing the 90 days limit and just keeping the 500 limit still be fine?

Reply:
> Removing entries from `m_attempt_tracker` when tx is removed from mempool doesn't seem necessary given `m_attempt_tracker` isn't used as a source for txs that need to be rebroadcasted.

I agree that rebroadcast would still work - the problem is more that `m_attempt_tracker` might not be able to fulfill its intended purpose with a 500 limit if it was keeping track of too many unnecessary transactions (and at the same time expel those that it was meant to keep track off after capacity is reached).

Reply:
I added commit 2ac22164f82a51d4eaece9acc42f346e8db679a5 to address this! I agree that when transactions are removed from the mempool for certain reasons, we are highly unlikely to see it again, so it makes sense for it to be removed from the rebroadcast attempt tracker. It's ok if it doesn't work perfectly for edge cases, but this behavior supports helps it work as intended. Thanks!